server/memphis_handler_schemas.go

```diff
@@ -194,21 +194,34 @@ func (sh SchemasHandler) CreateNewSchema(c *gin.Context) {
 		Versions: []primitive.ObjectID{newSchemaVersion.ID},
 	}
 
-	_, err = schemasCollection.InsertOne(context.TODO(), newSchema)
+	filter := bson.M{"name": newSchema.Name}
+	update := bson.M{
+		"$setOnInsert": bson.M{
+			"_id":  newSchema.ID,
+			"type": newSchema.Type,
+		},
+		"$set": bson.M{"versions": newSchema.Versions},
+	}
+	opts := options.Update().SetUpsert(true)
+	updateResults, err := schemasCollection.UpdateOne(context.TODO(), filter, update, opts)
 	if err != nil {
 		serv.Errorf("CreateSchema error: " + err.Error())
 		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
 		return
 	}
+	if updateResults.MatchedCount == 0 {
+		message := "Schema " + schemaName + " has been created"
```

Comment:
Here you should create the schema version otherwise it can be called more than once in a case of parallel requests